rpcs3/Emu/RSX/VK/VKGSRender.cpp

```diff
@@ -2412,20 +2413,20 @@ void VKGSRender::close_and_submit_command_buffer(vk::fence* pFence, VkSemaphore
 		primary_submit_info.wait_on(wait_semaphore, pipeline_stage_flags);
 	}
 
-	auto& async_scheduler = g_fxo->get<vk::AsyncTaskScheduler>();
-	if (async_scheduler.is_recording())
+	if (auto async_scheduler = g_fxo->try_get<vk::AsyncTaskScheduler>();
```

Comment:
Pretty sure the task scheduler should always exist, at least with the current code model. I'll have to check this properly when I have the time, otherwise it can cause nasty GPU hangs.

Reply:
No. it depends on the async compute capability

Reply:
Which I don't have for some reason

Reply:
ah. it needs async texture streaming. Otherwise the object is not created.

Reply:
Only if supports_asynchronous_compute property is true it is created, I believe this is mergeable because its needed for more changes.